cadwyn/applications.py

```diff
@@ -289,6 +290,7 @@ async def openapi_jsons(self, req: Request) -> JSONResponse:
                 routes=routes,
                 tags=self.openapi_tags,
                 servers=self.servers,
+                webhooks=self.webhooks.routes,
```

Comment:
It's going to be a bit more complex (as you highlighted in the issue). Let's say `User` model lost a field in a new version -- this means we will need its webhook to also lose that field in a new version, so we must either:

* version webhooks as well
* introduce a concept of "versioned" and "unversioned" webhooks which feels wrong to me and which I really wouldn't want to do. I am actually planning to remove unversioned endpoints too soon

And because we version webhooks as well, we will need to get another version of their routes. Let's think about how we could achieve this:

1. We need some point when we would generate them. I propose to use a [lifespan](https://fastapi.tiangolo.com/advanced/events/) that will be automatically attached to any other lifespans the user passes. It will run at the startup of the app and automatically generate all versioned webhook routes. I am planning to switch to the same interface for routes soon
2. To generate them, you can simply call `route_generation.generate_versioned_routers` on your webhook router
3. Then include something like the code below into the method that gets openapi:

```python
if version in self.router.versioned_routers:
            routes = self.router.versioned_routers[version].routes
            webhook_routes = self.router.versioned_webhook_routers[version].routes
            formatted_version = version.isoformat()
        elif version == "unversioned" and self._there_are_public_unversioned_routes():
            routes = self.router.unversioned_routes
            formatted_version = "unversioned"
        else:
            raise not_found_error

        # Add root path to servers when mounted as sub-app or proxy is used
        urls = (server_data.get("url") for server_data in self.servers)
        server_urls = {url for url in urls if url}
        root_path = self._extract_root_path(req)
        if root_path and root_path not in server_urls and self.root_path_in_servers:
            self.servers.insert(0, {"url": root_path})

        return JSONResponse(
            get_openapi(
                title=self.title,
                version=formatted_version,
                openapi_version=self.openapi_version,
                description=self.description,
                summary=self.summary,
                terms_of_service=self.terms_of_service,
                contact=self.contact,
                license_info=self.license_info,
                routes=routes,
                tags=self.openapi_tags,
                servers=self.servers,
                webhooks=webhook_routes,
            )
        )
```